### ISSUE TITLE
--no-mov-flow doesn't work

### ISSUE THREAD

**Issue by Smertig** [xoreaxeaxeax/movfuscator#19]
Hello!

Option `--no-mov-flow` seems not to work. I got these output:

```
M/o/Vfuscation complete.

/tmp/lcc61831.s: Assembler messages:
/tmp/lcc61831.s:1254: Error: too many memory references for `cmp'
/tmp/lcc61831.s:3293: Error: too many memory references for `cmp'
/tmp/lcc61831.s:5042: Error: too many memory references for `cmp'
/tmp/lcc61831.s:6292: Error: too many memory references for `cmp'
/tmp/lcc61831.s:9089: Error: too many memory references for `cmp'
/tmp/lcc61831.s:10983: Error: too many memory references for `cmp'
/tmp/lcc61831.s:12366: Error: too many memory references for `cmp'
```

To reproduce this bug, you can simply add `-Wf--no-mov-flow` at the end of 22nd line of `check.sh`

**Comment by GregoryMorse** [comment]
Change all the lines with:
print("cmpl "); emit_kid(p,1,nt); print(", "); emit_kid(p,0,nt); print("\n");
To:
print("movl "); emit_kid(p,0,nt); print(", %%eax\n");
print("cmpl "); emit_kid(p,1,nt); print(", %%eax\n");

Secondly, unfortunately the C libraries are precompiled when you build movfuscator.  emit_start is called in each build, but the result discarded.  The library version is what is used.  But unfortunately that is using the mov_flow flag, the default.  This is one way to do the job by changing default in c code, or much better is to change build.sh to pass --no-mov-flow to the libraries when they are precompiled.
static int mov_flow=        0;
Better approach in build.sh:
"$BUILDDIR/movcc" movfuscator/crt*.c -o "$BUILDDIR/crt*.c" -c -Wf--crt* -Wf--q -Wf--no-mov-flow where * is 0, f and d.

**Comment by GregoryMorse** [comment]
"cmpl (%s), 1\n" should be changed to "compl (%s), $1\n" also in cc_branch.  Looks as if this code was never tested...

**Comment by xoreaxeaxeax** [comment]
Thanks for the fixes, I'll get this committed shortly.  That flag worked at one point, but seems to have broken along the way.

**Comment by MrAureliusR** [comment]
Any chance this will get fixed at some point? Having the same problem still.

**Comment by xoreaxeaxeax** [comment]
Fixed with 6342ae76b580c174eb35e612ec9fc0013be7859d.  Thanks @GregoryMorse for the change suggestions.
